src/Fusion.Resources.Functions.Common/ApiClients/SummaryApiClient.cs

```diff
@@ -0,0 +1,35 @@
+using System.Text.Json;
+using Fusion.Resources.Functions.Common.Integration.Http;
+
+namespace Fusion.Resources.Functions.Common.ApiClients;
+
+
+public class SummaryApiClient : ISummaryApiClient
+{
+    private readonly HttpClient summaryClient;
+
+    public SummaryApiClient(IHttpClientFactory httpClientFactory)
+    {
+        summaryClient = httpClientFactory.CreateClient(HttpClientNames.Application.Summary);
+        summaryClient.Timeout = TimeSpan.FromMinutes(2);
+    }
+
+    public async Task PutDepartmentsAsync(IEnumerable<ApiResourceOwnerDepartments> departments,
+        CancellationToken cancellationToken = default)
+    {
+        var parallelOptions = new ParallelOptions()
+        {
+            CancellationToken = cancellationToken,
+            MaxDegreeOfParallelism = 10,
+        };
+
+        await Parallel.ForEachAsync(departments, parallelOptions, async (ownerDepartments, token) =>
+        {
+            var body = new JsonContent(JsonSerializer.Serialize(ownerDepartments));
+
+            // Error logging is handled by http middleware => FunctionHttpMessageHandler
+            using var _ = await summaryClient.PutAsync($"departments/{ownerDepartments.DepartmentSapId}", body,
+                cancellationToken);
```

Comment:
I used Parallel.ForEachAsync to easily limit concurrency, unsure what is preferred practice. Another method could be to chunk the departments and use Task.WhenAll for each chunk.

Alternatively, allow for sending a batch to the endpoint instead